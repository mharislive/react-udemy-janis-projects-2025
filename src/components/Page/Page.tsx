type PageProps = {
  children: React.ReactNode;
};

const Page = ({ children }: PageProps) => {
  return (
    <main className="flex bg-blue-950 text-gray-300 min-h-dvh">
      <div className="container max-w-5xl mx-auto py-10">{children}</div>
    </main>
  );
};

export default Page;
