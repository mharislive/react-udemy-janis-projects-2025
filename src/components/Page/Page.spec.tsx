import { render, screen } from "@testing-library/react";
import Page from "./Page";

describe("Page", () => {
  const PageComponent = () => (
    <Page>
      <h1>Hello World</h1>
    </Page>
  );

  it("should render successfully", () => {
    render(<PageComponent />);
     expect(
       screen.getByRole("heading", { name: /hello world/i })
     ).toBeInTheDocument();
  });

  it("should render passed children successfully", () => {
    render(<PageComponent />);
    expect(screen.getByText(/hello world/i)).toBeVisible();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<PageComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
