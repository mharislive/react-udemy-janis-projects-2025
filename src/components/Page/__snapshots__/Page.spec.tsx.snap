// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page > should match snapshot 1`] = `
<DocumentFragment>
  <main
    class="flex bg-blue-950 text-gray-300 min-h-dvh"
  >
    <div
      class="container max-w-5xl mx-auto py-10"
    >
      <h1>
        Hello World
      </h1>
    </div>
  </main>
</DocumentFragment>
`;
